docs/testing/e2e_debugging.md

```diff
@@ -0,0 +1,120 @@
+# How to debug an E2E test for Kubeflow Training Operator
+
+[E2E Testing](./e2e_testing.md) gives an overview of writing e2e tests. This guidance concentrateso more on the e2e failure debugging.
+
+
+## Prerequsite
+
+1. Install python 3.7
+
+2. Clone `kubeflow/testing` repo under `$GOPATH/src/kubeflow/`
+
+3. Install [ksonnet](https://ksonnet.io/)
+
+```
+wget https://github.com/ksonnet/ksonnet/releases/download/v0.13.1/ks_0.13.1_linux_amd64.tar.gz
+tar -xvzf ks_0.13.1_linux_amd64.tar.gz
+sudo cp ks_0.13.1_linux_amd64/ks /usr/local/bin/ks-13
+```
+> We would like to deprecate `ksonnet` but may takes some time. Feel free to pick up [the issue](https://github.com/kubeflow/training-operator/issues/1468) if you are interested in it. 
+> If your platform is darwin or windows, feel free to download binaries in [ksonnet v0.13.1](https://github.com/ksonnet/ksonnet/releases/tag/v0.13.1)
+
+4. Deploy HEAD training operator version in your environment
+
+```
+IMG=kubeflow/training-operator:e2e-debug-prid make docker-build
+
+# Optional - load image into kind cluster if you are using kind
+kind load docker-image kubeflow/training-operator:e2e-debug-1462
+
+kubectl set image deployment.v1.apps/training-operator training-operator=kubeflow/training-operator:e2e-debug-1462
+```
+
+## Run E2E Tests locally
+
+1. Set environments
+```
+export KUBEFLOW_PATH=/home/jiaxin.shan/go/src/github.com/kubeflow
```

Comment:
Let's use `$HOME`?

Reply:
I forget to make the change here. should be $GOPATH/src/

Reply:
resolved